### PR TITLE
HIVE-28271: DirectSql fails for AlterPartitions.

### PR DESCRIPTION
**Pull request by ayushtkn**
### What changes were proposed in this pull request?

Fix DirectSql for AlterPartitions for some DB

### Why are the changes needed?

To fix directSql for alter partions

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?

No

### How was this patch tested?

Checked Logs to see no direct sql related exception where they were present earlier

### CODE REVIEW

**Comment by ayushtkn** [comment]
cc. @wecharyu

**Comment by ayushtkn** [comment]
the test failures seems related:
http://ci.hive.apache.org/job/hive-precommit/job/PR-5256/1/testReport/junit/org.apache.hadoop.hive.metastore.client/TestAlterPartitions/

I think earlier directSql wasn't running so they slipped through, not sure if it is a test issue or actual prod issue, the tests passes individually but fail when run as entire class

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java [hunk 1]
```diff
@@ -548,6 +548,16 @@ public void addPartitions(List<MPartition> parts, List<List<MPartitionPrivilege>
    */
   public List<Partition> alterPartitions(MTable table, List<String> partNames,
                                          List<Partition> newParts, String queryWriteIdList) throws MetaException {
+
+    for (Partition tmpPart : newParts) {
```
Comment (wecharyu) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java:552]: Can we move this common check into `ObjectStore#alterPartitions`?

**Comment by wecharyu** [comment]
LGTM overall, only a nit code change request. BTW the error mainly comes from Derby DB right? @ayushtkn

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java (hunk 1 above)
Comment (ayushtkn) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java]: done

**Comment by ayushtkn** [comment]
Should be Oracle DB as well, they also return clob

**Review by wecharyu — APPROVED** [review]

**Review by zhangbutao — APPROVED** [review]
LGTM +1

**Pull request merged**